appscripts/canalytics.gs

```diff
@@ -145,40 +153,212 @@ function scoreBreakdownSummaryHeader(){
+        summaryObjects[scoutingItem['Team Number']]['Avg_Teleop_GPs'] += teleopGP
+
+        if (scoutingItem['Endgame Position'] == "PARK" ){
+          endgameScore +=2
+        }
+        if (scoutingItem['Endgame Position'] == "DOCK" ){
+          endgameScore +=6
+        }
+        if (scoutingItem['Endgame Position'] == "ENGAGED" ){
+          endgameScore +=10
+        }
+        summaryObjects[scoutingItem['Team Number']]['Avg_Endgame_Score'] += endgameScore
+      }
+
+      if ( summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] == "none" ) {
+        if (scoutingItem['Pieces Scored on Mid/High'] == "BOTH" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+        if (scoutingItem['Pieces Scored on Mid/High'] == "CUBES" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "cubes"
+          }
+        if (scoutingItem['Pieces Scored on Mid/High'] == "CONES" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "cones"
+          }
+      }
+      if ( summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] == "cones" ) {
+        if (scoutingItem['Pieces Scored on Mid/High'] == "BOTH" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+        if (scoutingItem['Pieces Scored on Mid/High'] == "CUBES" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+      }
+      if ( summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] == "cubes" ) {
+        if (scoutingItem['Pieces Scored on Mid/High'] == "BOTH" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+        if (scoutingItem['Pieces Scored on Mid/High'] == "CONES" ){
+            summaryObjects[scoutingItem['Team Number']]['Pieces_Scored'] = "both"
+          }
+      }
+   })  
+  Object.keys(summaryObjects).forEach( k => {
+    if ( summaryObjects[k]['Matches_Played'] > 0 ) {
+      summaryObjects[k]['Avg_Auton_Score'] = summaryObjects[k]['Avg_Auton_Score']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Auton_GPs'] = summaryObjects[k]['Avg_Auton_GPs']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Teleop_Score (no links)'] = summaryObjects[k]['Avg_Teleop_Score (no links)']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Teleop_GPs'] = summaryObjects[k]['Avg_Teleop_GPs']/summaryObjects[k]['Matches_Played']
+      summaryObjects[k]['Avg_Endgame_Score'] = summaryObjects[k]['Avg_Endgame_Score']/summaryObjects[k]['Matches_Played']
+    }
+  })
+
+  return summaryObjects
+}
+
+function specificScoutingSummaryHeader(){
+  return ["Auton Score", "Teleop Score", "Endgame Score", "Total Score"]
+}
+
+function specificScoutingSummary(specificTeamObjects, specificScoutingObjects){
+  Object.keys(specificTeamObjects).forEach( k => {
```

Comment:
specificTeamObjects is the output here, I think.  So, in the beginning of this routine, it should be empty.

If you log inside here, you will see only one loop through.

Reply:
Looking below, we might not need this loop, and instead initialize each specificScouting item a little later.